numpy/_core/tests/test_nditer.py

```diff
@@ -1126,7 +1126,12 @@ def test_iter_object_arrays_conversions():
         for x in i:
             x[...] += 1
     if HAS_REFCOUNT:
-        assert_(sys.getrefcount(ob) == rc - 1)
+        # TODO: why did this change?
+        newrc = sys.getrefcount(ob)
+        if sys.version_info < (3, 14):
+            assert_(newrc == rc - 1)
+        else:
+            assert_(newrc == rc)
```

Comment:
I don't understand why this changed - maybe @seberg or @mhvk might have some insight?

Reply:
I'm even confused why it worked. I guess the idea is that once you take out `ob` you have a new reference to it, as well as the existing one in the buffer array. Then, the in-place addition means that the object in the buffer array is replaced with a new object, so you get one less.

I guess on 3.14, the reference count does not get increased by taking the object out? I fear I'm as confused as you...  Maybe some inspection along the way can help?

Reply:
This has something to do with finalizers and `NpyIter_Close`. If I move the `HAS_REFCOUNT` check inside the `with` block, things work consistently. See f6ce63ba74.

@mattip it looks like you added the iterator-as-context-manager feature back in 2018. Maybe you have context about whether this refcounting behavior change is something to worry about.